### ISSUE TITLE
Signal rejected because of Android Auto "voice commands"

### ISSUE THREAD

**Issue by moxie0** [WhisperSystems/Signal-Android#6124]
```
Hi Developers at Open Whisper Systems,

Thanks for submitting your app for Android Auto. We reviewed your app, Signal Private Messenger, with package name org.thoughtcrime.securesms, and noticed some eligibility issues.

App status: Your app has not been accepted into Android Auto, and the update is not live on Google Play.

Eligibility issues by APK version:
APK Version(s)	Eligibility Issue
230	Your app does not support all of the required voice commands.
You can read more about how to support Voice Actions here: https://developer.android.com/training/auto/audio/index.html#support_voice
To be considered for Android Auto:

Make the necessary changes to your app.
Increment the version number of the upgraded APK.
Submit your app for another review.
Or, if you’d like to exclude your app from Android Auto:

Remove the Android Auto manifest entry from your APK code.
Sign in to your Developer Console and upload the new APK.
Click Submit update.
```

I don't really know what voice commands they're referring to, and I don't have any way to test.  It's blocking the release, so I'm going to remove android auto from the manifest entry for now. @snaggen are you willing to look into this?

**Comment by snaggen** [comment]
I seriously don't see anything there relevant for a messaging app. The link they post is to the page "Providing Audio Playback for Auto" page, and the Voice Comands related to that. 
If you look at the next page https://developer.android.com/training/auto/messaging/index.html
"Providing Messaging for Auto", we provide a complete implementation of that... including the Reply intent, which is the Voice Command for replying. 
I'm leaning towards a bad review in this case... if not, I really need more detailed information about what voice commands they want.

**Comment by moxie0** [comment]
From another email:

```
Hi Developers at Open Whisper Systems,

Thanks for submitting your app for Android Auto. We accepted Signal Private Messenger, with package name org.thoughtcrime.securesms, into the Android Auto program on Google Play with a Provisional Pass.

During review, we noticed the following issues that would usually fail for a production state app:

- When testing for voice commands, app does not respond to "Reply." Instead app replies with, "This chat app is either not installed or needs to be updated."

Version Name: 3.28.2
Version code: 230
Current track: BETA

Please correct these issues before re-submitting your app.

For more details on how an app can qualify for Android Auto, you can review the Android Auto App Quality Guidelines.

Thanks for supporting Google Play,

The Google Play Team
```

@snaggen You're saying the "reply" voice command does work?

**Comment by snaggen** [comment]
Well, The Reply intent works, since there besides the Voice Command is a default reply action attached to the notification that uses the Reply intent. So I have been able to test it that way. 
However, since Android Auto is not supported in Sweden, no voice commands work here.

But all the Intents from the "Providing Messaging" are implemented and tested one way or another.

**Comment by snaggen** [comment]
Can this be due to our Reply Receiver not being exported? That is the only difference I see from the docs regarding the reply action? 

 <receiver android:name=".notifications.AndroidAutoReplyReceiver"
             android:exported="false">
         <intent-filter>
           <action android:name="org.thoughtcrime.securesms.notifications.ANDROID_AUTO_REPLY"/>
         </intent-filter>

**Comment by snaggen** [comment]
I also note that we do not use 
`intent.addFlags(Intent.FLAG_INCLUDE_STOPPED_PACKAGES);`

as in the examples. 
```
Intent msgReplyIntent = new Intent()
    .addFlags(Intent.FLAG_INCLUDE_STOPPED_PACKAGES)
    .setAction("com.myapp.messagingservice.MY_ACTION_MESSAGE_REPLY")
    .putExtra("conversation_id", thisConversationId);

PendingIntent msgReplyPendingIntent = PendingIntent.getBroadcast(
    getApplicationContext(),
    thisConversationId,
    msgReplyIntent,
    PendingIntent.FLAG_UPDATE_CURRENT);
```

I have a patch in my aa_voice branch, if you think that might affect things.

**Comment by jcsp101** [comment]
@snaggen not sure if I can help somehow, but the "reply" action does not work for me at all on Android Auto for Signal (set as my SMS app, and doesn't work neither for SMS nor for Signal messages)

**Comment by snaggen** [comment]
I got in contact with a guy at Google, who told me that I should be able to get Voice commands to work if I just switch my device to US English... it shouldn't be tied to some specific region, just the language. So I will try this and see if I can get it to work. I will be busy the next couple of days though, so I might not be able to do it right away.

**Comment by FeuRenard** [comment]
Fixed by #6180 / 0f820e6

**Comment by coldwaterq** [comment]
I'm running into this issue now. I have the public version of the app from the play store. It is set as my default messaging app, and the dictated replies show as being sent from the messenger app although they never go out, so this may just be user error on my side. Let me know if I can help troubleshoot though.
